Sources/Endpoint.swift

```diff
@@ -20,9 +20,9 @@ public struct APIEndpoint<T> {
     let parameters: [String: Any]?
     
     /// The body to send with the request. Can be `nil`.
-    let body: Data?
+    let body: AnyEncodable?
     
-    init(path: String, method: HTTPMethod = .get, parameters: [String: Any]? = nil, body: Data? = nil) {
+    init(path: String, method: HTTPMethod = .get, parameters: [String: Any]? = nil, body: AnyEncodable? = nil) {
```

Comment:
Thinking out loud; what if we introduced a second generic parameter for the body? That way, we can still use `Encodable` and we don't need the `AnyEncodable`. I wonder whether this makes it too complicated but it might be worth the try!

Reply:
@AvdLee Apologies for not being clearer on my rationale behind `AnyEncodable`. The new generic parameter was actually my original thinking but started complicating things very early. I went ahead and implemented this approach now throughout the project (didn't really have another chance 😅) as it was a fun and interesting exercise anyway 😀! Some of the challenges that came up were:

1. The generic parameter `Request` will still have to be defined for GET requests, which feels somewhat unnatural. It can be defined as `Void`, however `Void` cannot conform to `Encodable`. As a workaround I introduced a dummy `EmptyBody` struct, that makes some endpoint definitions read as e.g.:
`extension APIEndpoint where Request == EmptyBody, Response == BundleIdBundleIdCapabilitiesLinkagesResponse`

2. I renamed the generic parameter `T` to `Response` on `APIEndpoint`. It would be a shame ending up with `APIEndpoint<Request: Encodable, Response>`, so I tweaked its definition to `APIEndpoint<Request: Encodable, Response: Decodable>` which reads quite nicely. However, this introduces similar issues as above on the `APIProvider`, since `Void` cannot conform to `Decodable`. This would happen for endpoints where the response type is irrelevant, and therefore it is marked as `Void`. Following a similar approach as for the request, I introduced a dummy type `IrrelevantResponse`. This makes endpoint definitions read as e.g.:
`extension APIEndpoint where Request == AppBetaTestersLinkagesRequest, Response == IrrelevantResponse` 
or 
`extension APIEndpoint where Request == EmptyBody, Response == IrrelevantResponse`

I understand these changes are quite substantial and involved some "workarounds" (perhaps I'm missing something obvious and we can avoid them... 🤔). Happy to drop some of these changes or the PR entirely if you feel there are too many trade-offs at this point.

Let me know what you think!

Reply:
I totally get the struggle, it was kind of expected once starting to dive into generics! We could, however, learn from Alamofire. They have an empty type defined: https://github.com/Alamofire/Alamofire/blob/755853f8fbf3b30c40e41cd5c9268bcbc599378a/Source/ResponseSerialization.swift#L676

With that, would it be more feasible to go with the generics approach?
It's common to have requests that return an empty response or a response we're not interested in so it's, either way, a good thing to built-in!

Reply:
Interesting, looks like I went with a very similar approach 🙂! Yes, definitely keen on keeping the generics approach. I'll wrap up the final bits soon a reach out for a final review. 
Thanks for digesting the long PR so far! 😀

Reply:
Thank you for thinking with me 💪 All for better code solutions and the learning process!